includes/html/api_functions.inc.php

```diff
@@ -1161,11 +1161,31 @@ function list_alert_rules(Illuminate\Http\Request $request)
     $sql = '';
     $param = [];
     if ($id > 0) {
-        $sql = 'WHERE id=?';
+        $sql = 'WHERE `a`.`id`=?';
         $param = [$id];
     }
 
-    $rules = dbFetchRows("SELECT * FROM `alert_rules` $sql", $param);
+    $rules = dbFetchRows("SELECT `a`.*, GROUP_CONCAT(DISTINCT `d`.`device_id`) AS `devices`, GROUP_CONCAT(DISTINCT `g`.`group_id`) AS `groups`, GROUP_CONCAT(DISTINCT `l`.`location_id`) AS `locations` FROM `alert_rules` AS `a` LEFT JOIN `alert_device_map` AS `d` ON `a`.`id`=`d`.`rule_id` LEFT JOIN `alert_group_map` AS `g` ON `a`.`id`=`g`.`rule_id` LEFT JOIN `alert_location_map` AS `l` ON `a`.`id`=`l`.`rule_id` $sql GROUP BY `a`.`id`", $param);
```

Comment:
could you write this on muliple lines instead? or better, convert it to eloquent

Reply:
I can write this on multiple lines.
But I spent some time trying to understand how to convert to Eloquent and I don't know how to do.
If it is acceptable with several lines I can update like this or if you really want to use Eloquent can you help me, please?
By the way in the api_functions.inc.php file, the other SQL queries are written as I first did.
And Eloquent is called downstream in the dbFetchRows function written in dbFacile.php file.
Thanks.